packages/manager/modules/web-paas/src/details/service/add-addon/routing.js

```diff
@@ -0,0 +1,37 @@
+import find from 'lodash/find';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('web-paas.dashboard.service.add-addon', {
+    url: '/:addonType',
+    views: {
+      modal: {
+        component: 'webPaasProjectAddStorage',
+      },
+    },
+    params: {
+      addonType: null,
+    },
+    layout: 'modal',
+    resolve: {
+      addonType: /* @ngInject */ ($transition$) =>
+        $transition$.params().addonType,
+      addon: /* @ngInject */ (WebPaas, project, addonType) => {
+        return WebPaas.getAdditionalOption(project.serviceId).then((result) => {
+          const addon = find(result, { productName: addonType });
```

Comment:
I do have a separate class for addon. There are kind of 3 different addon objects(from 3 different APIs). So i have added only to one of it, so that it doesnt get too confusing.